rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/NotificationAndIndicationManager.java

```diff
@@ -64,122 +62,85 @@
         return Observable.defer(new Func0<Observable<Observable<byte[]>>>() {
             @Override
             public Observable<Observable<byte[]>> call() {
-                synchronized (NotificationAndIndicationManager.this) {
+                synchronized (activeNotificationObservableMap) {
                     final CharacteristicNotificationId id
                             = new CharacteristicNotificationId(characteristic.getUuid(), characteristic.getInstanceId());
 
-                    final HashMap<CharacteristicNotificationId, Observable<Observable<byte[]>>> conflictingServerInitiatedReadingMap
-                            = withAck ? notificationObservableMap : indicationObservableMap;
-                    final boolean conflictingNotificationIsAlreadySet
-                            = conflictingServerInitiatedReadingMap.containsKey(id);
-
-                    if (conflictingNotificationIsAlreadySet) {
-                        return Observable.error(new BleConflictingNotificationAlreadySetException(characteristic.getUuid(), !withAck));
-                    }
-
-                    final HashMap<CharacteristicNotificationId, Observable<Observable<byte[]>>> sameNotificationTypeMap
-                            = withAck ? indicationObservableMap : notificationObservableMap;
-
-                    final Observable<Observable<byte[]>> availableObservable = sameNotificationTypeMap.get(id);
+                    final ActiveCharacteristicNotification activeCharacteristicNotification = activeNotificationObservableMap.get(id);
 
-                    if (availableObservable != null) {
-                        return availableObservable;
+                    if (activeCharacteristicNotification != null) {
+                        if (activeCharacteristicNotification.isIndication == withAck) {
+                            return activeCharacteristicNotification.notificationObservable;
+                        } else {
+                            return Observable.error(new BleConflictingNotificationAlreadySetException(characteristic.getUuid(), !withAck));
+                        }
                     }
 
                     final byte[] enableNotificationTypeValue = withAck ? configEnableIndication : configEnableNotification;
 
-                    final Observable<Observable<byte[]>> newObservable = createTriggeredReadObservable(
-                            characteristic,
-                            enableNotificationTypeValue,
-                            setupMode
-                    )
+                    final Observable<Observable<byte[]>> newObservable = setCharacteristicNotification(bluetoothGatt, characteristic, true)
+                            .compose(setupModeTransformer(descriptorWriter, characteristic, enableNotificationTypeValue, setupMode))
+                            .andThen(ObservableUtil.justOnNext(observeOnCharacteristicChangeCallbacks(gattCallback, id)))
```

Comment:
`ObservableUtil.justOnNext(T item)` == `Observable.never().startWith(T item)`
I do not think `Observable.just(T item)` would be enough here. Could you clarify?